src/controller/CHIPDevice.h

```diff
@@ -353,7 +353,12 @@ class DLL_EXPORT Device : public Messaging::ExchangeDelegate, public SessionEsta
 
     CASESession & GetCASESession() { return mCASESession; }
 
-    CHIP_ERROR GenerateCSRNonce() { return Crypto::DRBG_get_bytes(mCSRNonce, sizeof(mCSRNonce)); }
+    CHIP_ERROR SetCSRNonce(Optional<ByteSpan> csrNonce)
```

Comment:
We don't need to pass `csrNonce` as Optional type. It can be changed to 
```
CHIP_ERROR SetCSRNonce(const ByteSpan & csrNonce);
```

The caller can do `csrNonce.Value()` if needed before passing it as parameter.

Reply:
Done.